### ISSUE TITLE
Wrapping meteoIO with AutoWIG

### ISSUE THREAD

**Issue by thiemot** [StatisKit/AutoWIG#62]
After some unsuccessful attempts to write a python-wrapper for the C++ meteoIO-library ( https://models.slf.ch/p/meteoio/ ) using SIP and Cython, we now start using AutoWIG. 

We will share our experience with the wrapping of a large library with AutoWIG here.

**Comment by pfernique** [comment]
First, 
* On which OS do you intend to generate wrappers ?
* Is the source code publicly versioned (**SVN** or **Git** on which host) ?

I did have a quick glance at this library and it seems to raise no major difficulties.
But, to be sure, how the source code is relatively to the following guidelines provided in the article (p. 11-14):
- [ ] write self-contained headers,
- [ ] use of smart pointers,
- [ ] use *C++* **STL** containers,
- [ ] derive from `std::exception`,
- [ ] prevent `static` and `const` overloading,
- [ ] use namespaces.

**Comment by thiemot** [comment]
- We intend to generate wrappers for windows and linux.

- The source code is publicly versioned using SVN and can be found here:
https://models.slf.ch/svn/meteoio/trunk

**Comment by pfernique** [comment]
The target platform is not really a problem. I just didn't try **AutoWIG** when used with 2 different OSes.
I do use a Linux so if you are using a Windows resulting wrappers could be different with the same process (not too much, but it can be).

I won't have much time during the next week and half, so if you can give it a try using examples given in the article and post any problem here, it would be perfect. Then, ranging from February 13 to February 17, I have a window where I can give you a hand (if you didn't succeeded and are available). Therefore, the more your preliminary work is advanced, the better chances of finishing it on February 17 !

I didn't update the documentation, but to install all tools required for using **AutoWIG** and compiling its generated wrappers, the best is to install `autowig-toolchain` with **Conda** with the following command line

```console
conda create -n autowig-toolchain autowig-toolchain -c statiskit
```

**Comment by thiemot** [comment]
Hi Pierre,

I finally managed to get cython running. And I'm quite happy with it. But of course there are some problems, for example if the library is changing also the wrapper has to be adjusted. Furthermore I'm still having problems to get cython running under windows. It's horrible :)
So I might have a look at autowig. Especially I'm interested to see how complicated the installation on windows is...

Regards,
Thiemo

**Comment by pfernique** [comment]
Ok, closing the issue for now. Re-open it when you will take a look a **AutoWIG**

**Comment by nicholas512** [comment]
@thiemot  have you had any success in creating a python wrapper for meteoIO with either AutoWIG or cython?

**Comment by thiemot** [comment]
Hi nick, I finally created a wrapper with cython. It worked and should be checked in. But it is not very satisfying. When the code changes you also have to adjust the wrapper. The automatic stuff I tried did not work properly. -- Diese Nachricht wurde von meinem Android Mobiltelefon mit GMX Mail gesendet.Am 16.09.20, 20:22 schrieb Nick <notifications@github.com>:

   
  @thiemot have you had any success in creating a python wrapper for meteoIO with either AutoWIG or cython?
